ui/hooks/usePolling.ts

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useRef } from 'react';
+
+type UsePollingOptions = {
+  callback?: (pollingToken: string) => (pollingToken: string) => void;
+  startPollingByNetworkClientId: (
+    networkClientId: string,
+    options: any,
+  ) => Promise<string>;
+  stopPollingByPollingToken: (pollingToken: string) => void;
+  networkClientId: string;
+  options?: any;
+};
+
+const usePolling = (usePollingOptions: UsePollingOptions) => {
+  const pollTokenRef = useRef<null | string>(null);
+  const cleanupRef = useRef<null | ((pollingToken: string) => void)>(null);
+  let isMounted = false;
+  useEffect(() => {
+    isMounted = true;
+
+    const cleanup = () => {
+      if (pollTokenRef.current) {
+        usePollingOptions.stopPollingByPollingToken(pollTokenRef.current);
+        cleanupRef.current?.(pollTokenRef.current);
+      }
+    };
+    // Start polling when the component mounts
+    usePollingOptions
+      .startPollingByNetworkClientId(
+        usePollingOptions.networkClientId,
+        usePollingOptions.options,
+      )
+      .then((pollToken) => {
+        pollTokenRef.current = pollToken;
+        cleanupRef.current = usePollingOptions.callback?.(pollToken) || null;
+        if (!isMounted) {
+          cleanup();
+        }
+      });
+
+    // Return a cleanup function to stop polling when the component unmounts
+    return () => {
+      isMounted = false;
+      cleanup();
+    };
+  }, [
+    usePollingOptions.networkClientId,
+    usePollingOptions.options &&
+      JSON.stringify(
+        usePollingOptions.options,
+        Object.keys(usePollingOptions.options).sort(),
+      ),
```

Comment:
Wondering if its possible to simplify above conditional.

Reply:
This logic is to ensure that if the `usePollingOptions.options` is present we can guarantee the order of the keys in the option object so that the hook doesn't rerun unnecessarily. I'm not sure if there is a better way to express that conditional.